src/activities/quizzes/attempts/QuizAttemptsEntity.js

```diff
@@ -146,29 +139,15 @@ export class QuizAttemptsEntity extends Entity {
 	 */
 
 	_generateOverallGradeCalculationTypeAction(calculationType) {
-		if (!this._entity) return;
-		const action = this._entity.getActionByName(Actions.quizzes.attempts.updateOverallGradeCalculationType);
+		const entity = this.getOverallGradeCalculationSubEntity();

```

Comment:
This is the main fix, all else is tidying up

Reply:
thanks :D